### ISSUE TITLE
Make consensus module plugable

### ISSUE THREAD

**Issue by raullenchai** [iotexproject/iotex-core#2705]
For non-consensus nodes, e.g., API nodes, they do not need to even turn on the consensus module. We should come up with a design to make it pluggable.

**Comment by CoderZhi** [comment]
https://github.com/iotexproject/iotex-core/blob/0796d378aaedf3b802fead0090a7f26669a0c37f/config/config.go#L184 it is already here

**Comment by raullenchai** [comment]
@xianhuawei - can we try on one API node to `disable the consensus module`?

**Comment by raullenchai** [comment]
@Liuhaai pls validate if the flag is effective or not

**Comment by Liuhaai** [comment]
> @Liuhaai pls validate if the flag is effective or not

Yes. If `System.Active` flag is set to false, rolldpos consensus failed to work.

**Comment by raullenchai** [comment]
@xianhuawei pls consider disabling this one for all our API nodes.
